ports/renesas-ra/boards/VK-RA4W1: New board.

mbedNoobNinja opened the pull request:
`ports/renesas-ra/boards/`**`VK-RA4W1`:** Adding new dev board
`ports/renesas-ra/boards:` Based on existing **RA4W1** processor
`ports/renesas-ra:` Part of existing **RA** family
`ports/renesas-ra/ra:` Bonus 2 drivers **`PWM`** & **`DAC`**.

[Tests.log](https://github.com/micropython/micropython/files/10899285/W1_Tests.log)

For now the bonus is enabled only for VK-RA4W1, but with minimal changes, they should work for the other too.
For a full workability compliance with the other RA bords, they definitely must be checked by someone from Renesas.(may be @TakeoTakahashi2020)

At some point a suitable test cases should appear in `tests\renesas-ra:`, but it draws too much effort for me (unaware how to prepare & write tests), but Renesas should know better.

mbedNoobNinja committed:
ports/renesas-ra/boards/VK-RA4W1: New board.

ports/renesas-ra/boards: Integration to the existing RA4W1 processor.
ports/renesas-ra: Integration to the existing RA family.
ports/renesas-ra/ra: PWM & DAC support only for the VK-RA4W1.

Signed-off-by: mbedNoobNinja <novoltage@gmail.com>

TakeoTakahashi2020 commented:
Thanks, @mbedNoobNinja .
* I tried building and faced following error happend due to missing `#endif` in vector_data.c.  I could build it by adding `#endif`. 
```
...
GEN build-VK-RA4W1/genhdr/qstr.i.last
boards/VK-RA4W1/ra_gen/vector_data.c:4: error: unterminated #if
    4 | #if VECTOR_DATA_IRQ_COUNT > 0
...
```
* Regarding the board directory name, @dpgeorge guided me before that "we use underscore instead of hyphen in the directory name".  So it would be necessary to change it to `VK_RA4W1`.

mbedNoobNinja commented:
Hi @TakeoTakahashi2020, I changed the name of the board, as you and @dpgeorge mentioned, and fixed the file in question **`vector_data.c`**. Following the Contributor Guidelines and more specifically the commit [one](https://github.com/micropython/micropython/blob/master/CODECONVENTIONS.md#code-auto-formatting), they reccomend running codeformat.py beautifier before Pull request. Which I did. It turned out, that tool damages the files which need to be altered. When it replaces tabs with spaces, if there is some spaces at the begining of the row, sometimes the content after those spaces is dropped and the code breaks at compilation stage. Most of the FSP generated files's rows begins with TABs, but spits spaces here and there and that obveously bugs the tool. Anyway, I somehow managed the both happy (compiler & codeformat.py) and now **`VK_RA4W1`** should be compilable.

TakeoTakahashi2020 commented:
Hello @mbedNoobNinja, Thanks. I could build successfully for all boards including VK_RA4W1.  I do not have this board, so did only build check.  

I sometimes faced codeformat issues too and confirmed workflow result of Github Actions to investigate the detail. 
Regarding bonus drivers, I condier to apply them for other boards in the future.  As for FSP API utilizing, thanks for your comment described in source code.If possible, like to use, but some pin conflicts for FSP configuration.

mbedNoobNinja commented:
Hello @TakeoTakahashi2020, Thank you for your time cheking the new board. If you send me an address, I could send you 3 sample boards ([VK-RA4W1](https://vekatech.com/VK-RA4W1_docs/brochures/VK-RA4W1%20Flyer%20R2.pdf), [VK-RA6M3](https://vekatech.com/VK-RA6M3_docs/brochures/VK-RA6M3%20Flyer%20R2.pdf) & [VK-RA6M5](https://vekatech.com/VK-RA6M5_docs/brochures/VK-RA6M5%20Flyer%20R2.pdf)). My plans are to make micropython ports for the other 2 boards, but I find difficulties making the MCU's family csv file (**`ra6m3_af.csv`** & **`ra6m5_af.csv`**) because they do not exist yet and I am unaware in what pin count & alt function variety they possess. (Sure, I probably can find that in a specified mcu's datasheet, but it seems to me, it is alot of work, It must have an easy way) Could you enlighten me how to generate or create them? Can FSP help me and do that for me?

TakeoTakahashi2020 commented:
Hello @mbedNoobNinja, Thank you for additional info.  I will send you an email.

TakeoTakahashi2020 commented:
Hello @mbedNoobNinja -San, 

![image](https://user-images.githubusercontent.com/74999249/220257569-a9087e61-4379-421c-930b-8510b2b188b3.png)

You can specify "VK-RA4W1" of the product name for this macro definition.

mbedNoobNinja commented:
Hi @TakeoTakahashi2020 -San,
I fixed **`MICROPY_HW_BOARD_NAME`**, now it is with hyphen, but this `mpconfigport.h` is stuborn, whatever I change, it is always in conflict. I tried moving `#define MICROPY_SOFT_TIMER_TICKS_MS uwTick` before and after my changes but this doesn't resolved it either.

projectgus commented:
This is an automated heads-up that we've just merged a Pull Request
that removes the STATIC macro from MicroPython's C API.

See https://github.com/micropython/micropython/pull/13763

A search suggests this PR might apply the STATIC macro to some C code. If it
does, then next time you rebase the PR (or merge from master) then you should
please replace all the `STATIC` keywords with `static`.

Although this is an automated message, feel free to @-reply to me directly if
you have any questions about this.
